### PR TITLE
bumping fhir core version

### LINKED ISSUE #6363: update fhir_core dependency to avoid vulnerability
Fhir_core needs to be bumped to 6.3.23 to avoid https://avd.aquasec.com/nvd/2024/cve-2024-45294/

### PR DESCRIPTION
**Pull request by TipzCM**
closes #6363

### CODE REVIEW

**Review by jmarchionatto — APPROVED** [review]

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_6_0/6317-fix-resource-duplication-for-composite-unique-sp-with-date-time-component.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 type: fix
 issue: 6317
-backport: 7.4.3
+backport: 7.4.4
```
Comment (volodymyr-korzh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_6_0/6317-fix-resource-duplication-for-composite-unique-sp-with-date-time-component.yaml:4]: This `backport: 7.4.3` should remain for this changelog. Looks like it was changed by mistake.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_4_3/6363-updating-fhir-core-dependency.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+type: fix
+issue: 6363
+title: "Updating fhir_core dependency to fix vulnerability."
```
Comment (volodymyr-korzh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_4_3/6363-updating-fhir-core-dependency.yaml:4]: Suggestion - add version of core library to changelog

```
---
type: fix
issue: 6363
backport: 7.4.4
This release bumps the org.hl7.fhir core dependency up to 6.3.23, which includes vulnerability fixes.
```

As I undrestand it is a backport from master ? If yes, it should be put in the same folder as in master (`7_6_0`)
but with added backport property:

`backport: 7.4.4`

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/packages/PackageInstallerSvcImpl.java [hunk 3]
```diff
@@ -598,7 +598,8 @@ private boolean isValidResourceStatusForPackageUpload(IBaseResource theResource)
 				return (statusTypes.get(0).getValueAsString().equals("requested"));
 			case "DocumentReference":
 			case "Communication":
-				return (!statusTypes.get(0).getValueAsString().equals("?"));
+				return (statusTypes.get(0).isEmpty()
+						|| !statusTypes.get(0).getValueAsString().equals("?"));
```
Comment (volodymyr-korzh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/packages/PackageInstallerSvcImpl.java:602]: Question: are these changes is a result of core library update ?

**Review by volodymyr-korzh — CHANGES_REQUESTED** [review]

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_6_0/6317-fix-resource-duplication-for-composite-unique-sp-with-date-time-component.yaml (hunk 1 above)
Comment (TipzCM) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_6_0/6317-fix-resource-duplication-for-composite-unique-sp-with-date-time-component.yaml:4]: oops. updated

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/packages/PackageInstallerSvcImpl.java (hunk 3 above)
Comment (TipzCM) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/packages/PackageInstallerSvcImpl.java:602]: yeah

**Review by volodymyr-korzh — APPROVED** [review]
Looks good, approved

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_4_3/6363-updating-fhir-core-dependency.yaml (hunk 2 above)
Comment (TipzCM) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_4_3/6363-updating-fhir-core-dependency.yaml]: updated; but not a backport

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_6_0/6317-fix-resource-duplication-for-composite-unique-sp-with-date-time-component.yaml (hunk 1 above)
Comment (TipzCM) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_6_0/6317-fix-resource-duplication-for-composite-unique-sp-with-date-time-component.yaml]: fixed

**Pull request merged**